golem_messages/message/base.py

```diff
@@ -0,0 +1,382 @@
+import calendar
+import datetime
+import enum
+import hashlib
+import logging
+import struct
+import time
+
+import golem_messages
+
+from golem_messages import datastructures
+from golem_messages import exceptions
+from golem_messages import serializer
+from golem_messages import settings
+
+logger = logging.getLogger('golem.network.transport.message')
+
+
+def verify_time(timestamp):
+    """ Verify message timestamp. If message is to old or has timestamp from
+    distant future raise TimestampError.
+
+    NOTE: This method deliberately ignores microseconds - precision=1s
+    """
+    now = datetime.datetime.utcnow()
+    try:
+        msgdt = datetime.datetime.utcfromtimestamp(timestamp)
+    except (TypeError, OSError, OverflowError, ValueError) as e:
+        logger.debug('Error parsing timestamp: %r', timestamp, exc_info=True)
+        raise exceptions.TimestampError(str(e))
+    delta = now - msgdt
+    delta_future = msgdt - now
+    logger.debug('msgdt %s Δ %s Δfuture %s', msgdt, delta, delta_future)
+    if delta > settings.MSG_TTL:
+        raise exceptions.MessageTooOldError()
+    if delta_future > settings.FUTURE_TIME_TOLERANCE:
+        raise exceptions.MessageFromFutureError()
+
+
+def _fake_sign(s):  # pylint: disable=unused-argument

```

Comment:
unused argument can be replaced with `_*`

Reply:
fixed